sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -889,8 +897,9 @@ public partial class HttpPipeline
     public static partial class HttpPipelineBuilder
     {
         public static Azure.Core.Pipeline.HttpPipeline Build(Azure.Core.ClientOptions options, params Azure.Core.Pipeline.HttpPipelinePolicy[] perRetryPolicies) { throw null; }
+        public static Azure.Core.Pipeline.HttpPipeline Build(Azure.Core.ClientOptions options, Azure.Core.Pipeline.HttpPipelinePolicy[] perCallPolicies, Azure.Core.Pipeline.HttpPipelinePolicy[] perRetryPolicies, Azure.Core.MessageClassifier messageClassifier) { throw null; }
         public static Azure.Core.Pipeline.DisposableHttpPipeline Build(Azure.Core.ClientOptions options, Azure.Core.Pipeline.HttpPipelinePolicy[] perCallPolicies, Azure.Core.Pipeline.HttpPipelinePolicy[] perRetryPolicies, Azure.Core.Pipeline.HttpPipelineTransportOptions transportOptions, Azure.Core.ResponseClassifier? responseClassifier) { throw null; }
```

Comment:
We don't have another way to allow access to the full message for classification.  We can't just take an error code string, because different services put the error code in a message header (Storage) vs the body (Most of Azure).  I'm open to alternative ideas, though.  If we really want just an AddClassifier(statusCode, errorCode, bool isError) API, I'm sure we can find a way to provide an extensibility point to customize the retrieval mechanism for the error code, but it didn't feel worth it.

Can you help me understand the drawback of exposing MessageClassifier, with only its TryClassify method?